src/provable/from-layout.ts

```diff
@@ -0,0 +1,563 @@
+import {
```

Comment:
this file used to be called `transaction-helpers.ts`. it is the core of our "derived JS from OCaml" infrastructure. it was extended by adding the `fromJSON` and `emptyValue` functions, and was made generic over `Field`